slsim/lens_pop.py

```diff
@@ -1,378 +1,70 @@
-from slsim.ParamDistributions.los_config import LOSConfig
-import os
-import pickle
-
+import sncosmo
 import numpy as np
-from astropy.table import Table
 
 from slsim.lens import Lens
+from typing import Optional, Union
+from astropy.cosmology import Cosmology
 from slsim.lens import theta_e_when_source_infinity
+from slsim.Sources.source_pop_base import SourcePopBase
+from slsim.ParamDistributions.los_config import LOSConfig
+from slsim.Deflectors.deflectors_base import DeflectorsBase
 from slsim.lensed_population_base import LensedPopulationBase
-from slsim.Pipelines.skypy_pipeline import SkyPyPipeline
 
 
 class LensPop(LensedPopulationBase):
     """Class to perform samples of lens population."""
 
     def __init__(
         self,
-        deflector_type="elliptical",
-        source_type="galaxies",
-        kwargs_deflector_cut=None,
-        kwargs_source_cut=None,
-        kwargs_quasars=None,
-        kwargs_quasars_galaxies=None,
-        variability_model=None,
-        kwargs_variability=None,
-        kwargs_mass2light=None,
-        skypy_config=None,
-        slhammocks_config=None,
-        sky_area=None,
-        source_sky_area=None,
-        deflector_sky_area=None,
-        filters=None,
```

Comment:
how do we make sure that the correct filters are chosen in the Deflector and Source classes? Could cause issues if they are not the same.

Reply:
Good catch! I'd add a ValueError in LensPop to catch this, since this is a requirement from LensPop and all source / deflector related attributes should be owned by their respective classes.

Reply:
Coming back to this, how willing are we to let this be up to the user? Currently the filters kwarg is only used in the case of galaxy-galaxy strong lensing generated via SkyPyPipeline. For other sources the user has to make sure that kwargs_source_cut selects the same filters as provided by the deflector pipeline.

I'd argue for leaving this be in the current PR and adding it as a task for #225 , where we can make the available filters an attribute, such that we can add a check in LensPop as mentioned in my previous comment.

Reply:
Hi @jacob-hjortlund the kwargs_source_cut is only when you want to make cuts in the distribution. SNe also need to be provided filters (and so do quasars). @nkhadka21 made an issue in the past to homogenize the filter naming among the different galaxy/point-source classes. Perhaps you can leave it as a #TODO in the initialization, or define a routine check_consistency() in LensPop that is generally operating on class instances of sources and deflectors and might help diagnose issues for the user to be raised

Reply:
In that case I'll add this as a TODO in the init, since source classes will be modified in #225 and onwards.